zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -1708,20 +1708,16 @@ public void onAdd(String interpreterGroupId, AngularObject object) {
 
   @Override
   public void onUpdate(String interpreterGroupId, AngularObject object) {
-    if (getNotebook() == null) {
+    if (getNotebook() == null || object.getNoteId() == null) {
       return;
     }
 
-    List<Note> notes = getNotebook().getAllNotes();
-    for (Note note : notes) {
-      if (object.getNoteId() != null && !note.getId().equals(object.getNoteId())) {
-        continue;
-      }
-
+    Note note = getNotebook().getNote(object.getNoteId());
```

Comment:
Judge `object.getNoteId() == null` Is to prevent `Note note = getNotebook().getNote(object.getNoteId());` exception

Reply:
I'm actually not sure why it wasn't doing `getNotebook().getNote(note id)` before @jongyoul @zjffdu ?

Reply:
`object.getNoteId() == null`, Can also not add this code, Just to be safer, Or do I cancel this judgment code?